### DISCUSSION TITLE
Subscribe to SDK Client AWS_MAX_ATTEMPTS retries

### DISCUSSION THREAD

**Discussion by skirk-mpr** [aws/aws-sdk-net#3340] [Q&A] [answered]
Is there anyway to gather data or otherwise hook into the built in retry mechanism (e.g. subscribe) for the SDK client(s), as to gather data about the frequency of the need to perform retries for individual requests to AWS Services?  For some services (e.g. DynamoDB/S3) it will be helpful to monitor the frequency in which we are hitting service limits.

**Comment by normj** [comment, marked as answer]
@skirk-mpr The retries are logged as part of the SDK logging mechanism. We are at the start of an overhaul in the SDK to rework telemetry component of the SDK. The first of many PRs was just put out, https://github.com/aws/aws-sdk-net/pull/3344. When that is complete you should be able to monitor the retries via Opentelemetry.
